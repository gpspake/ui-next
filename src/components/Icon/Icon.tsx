import React from 'react';
import classNames from 'classnames';
import styles from './icon.scss';
import icons from './icon-config';

type IconProps = {
  /**
   * Name of icon to render.
   */
  icon: keyof typeof icons,
  /**
   * Pass in custom styling like style={{padding: 10px}}
   */
  customStyling?: {},
  /**
   * Listen for icon click event and pass back event
   */
  onClick?: React.MouseEventHandler<HTMLButtonElement>,
  /**
   * Listen for icon focus event and pass back event
   */
  onFocus?: React.FocusEventHandler<HTMLInputElement>,
  /**
   * Listen for icon onMouseDown event and pass back event
   */
  onMouseDown?: React.MouseEventHandler<HTMLButtonElement>,
  /**
   * Additional className
   */
  className?: string,
  /**
   * icon color
   */
  color?: 'currentColor'|
  'dark'|
  'light'|
  'primary'|
  'primary-hc'|
  'secondary'|
  'tertiary',

  /**
   * Rotate the icon by degrees [90, 180, 270]
   */
  rotate?: 0 | 90 | 180 | 270
  /**
   * one of [xsmall: .5em, small: .75em, normal: 1em, medium: 1.5em, large: 2em]
   */
  size?: 'xsmall'| 'small'| 'normal'| 'medium'| 'large'
};

/**
 * A component for rendering UI icons
 */
export const Icon = (props: IconProps) => {

  const {
    icon,
    color = 'currentColor',
    className = null,
    rotate = null,
    size = 'normal',
    onClick = () => {},
    onFocus = () => {},
    onMouseDown = () => {},
    customStyling = {}
  } = props

  return (
    <span
      className={classNames([
        styles.base,
        styles[color],
        styles[size],
        className
      ])}
      dangerouslySetInnerHTML={{ __html: icons[icon] }}
      style={{
        transform: rotate ? `rotateZ(${rotate}deg)` : '',
        ...customStyling
      }}
      onClick={onClick}
      onFocus={onFocus}
      onMouseDown={onMouseDown}
      role="presentation"
    />
  );
}

Icon.displayName = 'UIIcon';

export default Icon;
