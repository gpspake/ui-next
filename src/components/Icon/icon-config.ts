// Icon Imports
import add from './icons/add.svg';
import administrativeArea from './icons/administrative-area.svg';
import arrow from './icons/arrow.svg';
import alert from './icons/alert.svg';
import barChart from './icons/bar-chart.svg';
import bestMobileAward from './icons/best-mobile-award.svg';
import fiveG from './icons/5g.svg'
import browser from './icons/browser.svg';
import calendar from './icons/calendar.svg';
import caret from './icons/caret.svg';
import checkCircle from './icons/checkCircle.svg';
import check from './icons/check.svg';
import chevron from './icons/chevron.svg';
import chip from './icons/chip.svg';
import chipsetManufacturer from './icons/chip-manufacturer.svg';
import city from './icons/city.svg';
import clearAll from './icons/clearAll.svg';
import close from './icons/close.svg';
import cloudDown from './icons/cloudDown.svg';
import cloudUp from './icons/cloudUp.svg';
import cma from './icons/cma.svg';
import code from './icons/code.svg';
import combined from './icons/combined.svg';
import company from './icons/company.svg';
import controller from './icons/controller.svg';
import country from './icons/country.svg';
import coverage from './icons/coverage.svg';
import customization from './icons/customization.svg';
import desktop from './icons/desktop.svg';
import duplicate from './icons/duplicate.svg';
import deviceManufacturer from './icons/device-manufacturer.svg';
import display from './icons/display.svg';
import download from './icons/download.svg';
import dualMaps from './icons/dualMaps.svg';
import edit from './icons/edit.svg';
import edit2 from './icons/edit2.svg';
import expand from './icons/expand.svg';
import eye from './icons/eye.svg';
import exportData from './icons/exportData.svg';
import facebook from './icons/facebook.svg';
import feedback from './icons/feedback.svg';
import filter from './icons/filter.svg';
import gauge from './icons/gauge.svg';
import gear from './icons/gear.svg';
import globe from './icons/globe.svg';
import glossary from './icons/glossary.svg';
import help from './icons/help.svg';
import home from './icons/home.svg';
import host from './icons/host.svg';
import horizontalStackedChart from './icons/horizontalStackedChart.svg';
import info from './icons/info.svg';
import isp from './icons/isp.svg';
import internet from './icons/internet.svg';
import jitter from './icons/jitter.svg';
import lineChart from './icons/line-chart.svg';
import list from './icons/list.svg';
import locality from './icons/locality.svg';
import location from './icons/location.svg';
import lock from './icons/lock.svg';
import lockOpen from './icons/lockOpen.svg';
import loss from './icons/loss.svg';
import mail from './icons/mail.svg';
import menu from './icons/menu.svg';
import map from './icons/map.svg';
import mailOpen from './icons/mailOpen.svg';
import maintenance from './icons/maintenance.svg';
import methodology from './icons/methodology.svg';
import mobile from './icons/mobile.svg';
import more from './icons/more.svg';
import mosaicGrid from './icons/mosaic-grid.svg';
import multiple from './icons/multiple.svg';
import multiple2 from './icons/multiple2.svg';
import neighborhood from './icons/neighborhood.svg';
import networkPrivate from './icons/networkPrivate.svg';
import networkPublic from './icons/networkPublic.svg';
import overview from './icons/overview.svg';
import packet from './icons/packet.svg';
import pc from './icons/pc.svg';
import pin from './icons/pin.svg';
import ping from './icons/ping.svg';
import play from './icons/play.svg';
import postalCode from './icons/postal-code.svg';
import ranking from './icons/ranking.svg';
import radioDot from './icons/radio-dot.svg';
import satellite from './icons/satellite.svg';
import satellite2 from './icons/satellite2.svg';
import save from './icons/save.svg';
import save2 from './icons/save2.svg';
import saveAs from './icons/saveAs.svg';
import search from './icons/search.svg';
import settings from './icons/settings.svg';
import selection from './icons/selection.svg';
import server from './icons/server.svg';
import schedule from './icons/schedule.svg';
import share from './icons/share.svg';
import share2 from './icons/share2.svg';
import signal from './icons/signal.svg';
import single from './icons/single.svg';
import singleList from './icons/single-list.svg';
import star from './icons/star.svg';
import starEmpty from './icons/starEmpty.svg';
import status from './icons/status.svg';
import subscription from './icons/subscription.svg';
import reload from './icons/reload.svg';
import tablet from './icons/tablet.svg';
import table from './icons/table.svg';
import topRatedCell from './icons/top-rated-cell.svg';
import topRatedFixed from './icons/top-rated-fixed.svg';
import trash from './icons/trash.svg';
import trash2 from './icons/trash2.svg';
import twitter from './icons/twitter.svg';
import upgrade from './icons/upgrade.svg';
import upload from './icons/upload.svg';
import user from './icons/user.svg';
import users from './icons/users.svg';
import verticalBarChart from './icons/vertical-bar-chart.svg';
import video from './icons/video.svg';
import wifi from './icons/wifi.svg';
import video5gMobile from './icons/video5gMobile.svg';
import videoFixed from './icons/videoFixed.svg';
import videoMobile from './icons/videoMobile.svg';

export default {
  fiveG,
  add,
  administrativeArea,
  arrow,
  alert,
  barChart,
  bestMobileAward,
  browser,
  calendar,
  caret,
  checkCircle,
  check,
  chevron,
  chip,
  chipsetManufacturer,
  city,
  clearAll,
  close,
  cloudDown,
  cloudUp,
  cma,
  code,
  combined,
  company,
  controller,
  country,
  coverage,
  customization,
  desktop,
  duplicate,
  deviceManufacturer,
  display,
  download,
  dualMaps,
  eye,
  edit,
  edit2,
  expand,
  exportData,
  facebook,
  feedback,
  filter,
  gauge,
  gear,
  globe,
  glossary,
  help,
  home,
  host,
  horizontalStackedChart,
  info,
  internet,
  isp,
  jitter,
  lineChart,
  list,
  locality,
  location,
  lock,
  lockOpen,
  loss,
  mail,
  menu,
  map,
  mailOpen,
  maintenance,
  methodology,
  mobile,
  more,
  mosaicGrid,
  multiple,
  multiple2,
  neighborhood,
  networkPrivate,
  networkPublic,
  overview,
  packet,
  pc,
  pin,
  ping,
  play,
  postalCode,
  ranking,
  radioDot,
  satellite,
  satellite2,
  save,
  save2,
  saveAs,
  search,
  settings,
  selection,
  server,
  schedule,
  share,
  share2,
  signal,
  single,
  singleList,
  star,
  starEmpty,
  status,
  subscription,
  reload,
  table,
  tablet,
  topRatedCell,
  topRatedFixed,
  trash,
  trash2,
  twitter,
  upgrade,
  upload,
  user,
  users,
  video,
  verticalBarChart,
  wifi,
  video5gMobile,
  videoFixed,
  videoMobile
};
