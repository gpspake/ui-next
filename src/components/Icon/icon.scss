.base {
    display: inline-flex;
    transition: transform 250ms ease;

& svg {
      display: block;
      width: 100%;
      height: auto;
  }
}

.xsmall {
    width: 0.5em;
    height: 0.5em;
}

.small {
    width: 0.75em;
    height: 0.75em;
}

.normal {
    width: 1em;
    height: 1em;
}

.medium {
    width: 1.5em;
    height: 1.5em;
}

.large {
    width: 2em;
    height: 2em;
}

.dark {
    color: var(--dark-gray);
}

.light {
    color: var(--white);
}

.primary {
    color: var(--blue);
}

.primary-hc {
    color: var(--dark-blue);
}

.secondary {
    color: var(--purple);
}

.tertiary {
    color: var(--purple);
}
