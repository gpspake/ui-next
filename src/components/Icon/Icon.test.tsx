import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

it('matches snapshot', () => {
  const component = renderer.create(<Icon icon="lock" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('sets a default color of currentColor', () => {
  const testRenderer = renderer.create(<Icon icon="lock" />);
  const testInstance = testRenderer.root;
  expect(testInstance.findByType('span').props.className).toBe('base currentColor normal');
});
