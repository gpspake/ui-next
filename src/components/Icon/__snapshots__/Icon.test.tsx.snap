// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  className="base currentColor normal"
  dangerouslySetInnerHTML={
    Object {
      "__html": Object {},
    }
  }
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="presentation"
  style={
    Object {
      "transform": "",
    }
  }
/>
`;
